lib/metrics/index.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+const PrometheusClient = require('./prometheus');
+const StatsDClient = require('./statsd');
+const LogClient = require('./log');
+const Metric = require('./metric');
+const normalizeName = require('./utils').normalizeName;
+
+const SUPPORTED_CLIENTS = {
+    PROMETHEUS: 'prometheus',
+    STATSD: 'statsd',
+    LOG: 'log'
+};
+
+const DEPRECATED_METHODS = ['timing', 'increment', 'decrement', 'gauge', 'unique'];
+
+class Metrics {
+    constructor(options, logger) {
+        this.logger = logger;
+        this.cache = new Map();
+        this.clients = [];
+        options.forEach((o) => {
+            switch (o.type) {
+                case SUPPORTED_CLIENTS.PROMETHEUS:
+                    this.clients.push(new PrometheusClient(o, this.logger));
+                    break;
+                case SUPPORTED_CLIENTS.STATSD:
+                    this.clients.push(new StatsDClient(o, this.logger));
+                    break;
+                case SUPPORTED_CLIENTS.LOG:
+                    this.clients.push(new LogClient(o, this.logger));
+                    break;
+                default:
+                    logger.log('error/metrics', `No such metrics client: '${o.type}'`);
+            }
+        });
+    }
+
+    fetchClient(name) {
+        return this.clients.find((client) => {
+            return client.constructor.name === name;
+        });
+    }
+
+// Example Options:
+// {
+//     type: 'Counter',
+//     name: 'hitcount',
+//     prometheus: {
+//         name: 'hitcount',
+//         help: 'hit count',
+//         buckets: [], // https://github.com/siimon/prom-client#histogram
+//         percentiles: [], // https://github.com/siimon/prom-client#summary
+//     },
+//     labels: {
+//         keys: [],
+//         labelPosition: 'before',
+//         omitLabelNames: false
+//     }
+// }
+    makeMetric(options) {
+        const metric = new Metric(this.clients, this.logger, options);
+        this.cache.set(options.name, metric);
```

Comment:
What if the metric already exists with different settings?

Reply:
If you mean a collision based on name, then it will silently overwrite it in the cache or depending on configuration throw an error from the metrics client itself.  Prometheus names get registered and a collision there will throw an error.  I'm open to suggestions here.